interp/decls.ml

```diff
@@ -62,6 +62,7 @@ type logical_kind =
 type variable_data = {
   opaque:bool;
   kind:logical_kind;
+  goal_visibility:bool;
```

Comment:
I am not sure this table is the right place to store this info, instead it seems to me that all of `Decls` should go away.

If the data has a lifetime similar to `Proof.t` info should live there I guess.